### PR TITLE
Update of VBF HLT filter

### PR DESCRIPTION
**Pull request by silviodonato**
In order to implement the new VBF trigger strategy ( https://indico.cern.ch/event/351559/contribution/1/11/material/slides/6.pdf ), we need to update the HLTJetSortedVBFFilter.

This PR add "1BTagAndEta" and "2BTagAndPt" modes to use HLTJetSortedVBFFilter.

The "1BTagAndEta"  defines as:
- b1, the most b-tagged jet
- q1,q1, the most forward and backward remaining jets
- b2, the remaining jet with higher pT

The "2BTagAndPt"  defines as:
- b1, b2 the two most b-tagged jets
- q1, q2 the remaining jets with higher pT

cc: @grauco , @pazzurri

### CODE REVIEW

File: HLTrigger/JetMET/interface/HLTJetSortedVBFFilter.h [hunk 1]
```diff
@@ -36,6 +36,9 @@ class HLTJetSortedVBFFilter : public HLTFilter {
   static bool comparator ( const Jpair& l, const Jpair& r) {
     return l.first < r.first;
   }
+  static bool comparator_inv ( const Jpair& l, const Jpair& r) {
+    return l.first > r.first;
+  }
```
Comment (Martin-Grunewald) [inline HLTrigger/JetMET/interface/HLTJetSortedVBFFilter.h]: comparator_inv is not uswed anywhere, so please remove these lines.
And if you want it in 73X (ie, before HLT moves to 74X), remember to make a 73X PR as well.

Comment (silviodonato) [inline HLTrigger/JetMET/interface/HLTJetSortedVBFFilter.h]: Thanks. I've updated the PR.

**Comment by perrotta** [comment]
It does not merge: please synchronize with the HEAD before re-submitting:

++<<<<<<< HEAD
 +  descriptions.add(defaultModuleLabel<HLTJetSortedVBFFilter<T>>(), desc);
++=======
+   desc.add<int>("njets",4);
+   descriptions.add(string("hlt")+string(typeid(HLTJetSortedVBFFilter<T>).name()),desc);
++>>>>>>> cms-sw/refs/pull/7079/head

Since you are on it: could you please also indent correctly all the new code?
Then, do the same also for the 73X backport.

**Comment by silviodonato** [comment]
@perrotta , I didn't solve the problem yet... Let me try again

**Comment by perrotta** [comment]
You must rebase starting from a more recent IB, e.g. CMSSW_7_4_X_2015-01-08-0200

File: HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc [hunk 2]
```diff
@@ -28,59 +28,65 @@ using namespace std;
...
-  desc.add<double>("Mqq",200);
-  desc.add<double>("Detaqq",2.5);
-  desc.add<double>("Detabb",10.);
-  desc.add<double>("Dphibb",10.);
-  desc.add<double>("Ptsumqq",0.);
-  desc.add<double>("Ptsumbb",0.);
-  desc.add<double>("Etaq1Etaq2",40.);
-  desc.add<std::string>("value","second");
-  desc.add<int>("triggerType",trigger::TriggerJet);
-  descriptions.add(defaultModuleLabel<HLTJetSortedVBFFilter<T>>(), desc);
```
Comment (Martin-Grunewald) [inline HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc]: This removal (and addition of
```
descriptions.add(string("hlt")+string(typeid(HLTJetSortedVBFFilter<T>).name()),desc);
```
is reverting a 74X change we need to keep!

Comment (silviodonato) [inline HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc]: I just add a "\t" instead of a space. 
Below you can see that I'm adding the same lines (plus "njets" description).

Comment (Martin-Grunewald) [inline HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc]: No, what I mean is that you effectively changed the one line from:
```
descriptions.add(defaultModuleLabel<HLTJetSortedVBFFilter<T>>(), desc);
```
to now in your PR to be:
```
descriptions.add(string("hlt")+string(typeid(HLTJetSortedVBFFilter<T>).name()),desc);
```
and this is not good.

Please revert back to the 
```
descriptions.add(defaultModuleLabel<HLTJetSortedVBFFilter<T>>(), desc);
```
in 74X (in 73X it is different!)

Comment (silviodonato) [inline HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc]: I missed the line. Thanks!

**Comment by Martin-Grunewald** [comment]
-1
Same crash also in 74X (as reported in 73X backport PR https://github.com/cms-sw/cmssw/pull/7081)

**Comment by Martin-Grunewald** [comment]
Why is this PR now changing 45 files? Please rebase so that only your files are actually modified.

**Comment by ianna** [comment]
+1

@silviodonato - if you rebase it, Geometry signature will not be needed, I think. The PR included in this one is already merged in 74x IB

**Comment by silviodonato** [comment]
Rebased with CMSSW_7_4_X_2015-01-12-0200.

**Comment by silviodonato** [comment]
Removed EventFilter/ESRawToDigi/python/esRawToDigiTmp_cfi.py (I don't understand why it was added)

**Comment by perrotta** [comment]
Please Silvio, do NOT remove files from the release, unless they are those belonging to the modifications you plan for this PR!
Please revert your last commit-

**Comment by silviodonato** [comment]
Thanks, I removed the last commit.
About the build test failure, I get error also with the clean version of  CMSSW_7_4_X_2015-01-12-0200.

**Comment by Martin-Grunewald** [comment]
+1
